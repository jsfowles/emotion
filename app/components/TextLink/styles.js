import glamorous from 'glamorous';

import { COLORS, ANIMATION } from '@identity/constants';

export const A = glamorous.a({
  color: COLORS.brandPrimary,
  textDecoration: 'none',
  transition: `color ${ANIMATION.durationShort} ${ANIMATION.timing}`,

  '&:hover': {
    color: COLORS.brandSecondary,
  },
});
