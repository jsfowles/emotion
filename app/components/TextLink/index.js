import React from 'react';
import PropTypes from 'prop-types';

import { A } from './styles';

const TextLink = ({ href, text, internal }) => (internal ? (
  null // TODO: Setup <Link />
) : (
  <A href={ href } target="_blank">{ text }</A>
));

TextLink.propTypes = {
  href: PropTypes.string,
  text: PropTypes.string,
  internal: PropTypes.bool,
};

TextLink.defaultProps = {
  href: '#',
  text: 'Link with no text',
  internal: false,
};

export default TextLink;
