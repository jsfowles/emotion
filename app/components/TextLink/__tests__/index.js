import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TextLink from '@components/TextLink';

const defaultProps = {
  href: 'https://underbelly.is',
  text: 'Underbelly',
  internal: false,
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <TextLink { ...newProps } />;
};

describe('TextLink', () => {
  it('renders with props', () => {
    const wrapper = shallow(generateComponent());
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders with default props', () => {
    const wrapper = shallow(<TextLink />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
