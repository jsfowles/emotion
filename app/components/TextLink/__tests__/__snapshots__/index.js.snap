// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink renders with default props 1`] = `
<glamorous(a)
  href="#"
  target="_blank"
>
  Link with no text
</glamorous(a)>
`;

exports[`TextLink renders with props 1`] = `
<glamorous(a)
  href="https://underbelly.is"
  target="_blank"
>
  Underbelly
</glamorous(a)>
`;
